### PR TITLE
1425 - tree-grid appendData was breaking rendering of existing rows

### LINKED ISSUE #1425: IdsDatagrid: TreeGrid appendData() adds new rows but breaks current ones and issue on scrollend trigger in treegrid
**Describe the bug**
1) Adding new rows in the treegrid using appendData() breaks the current rows. Current rows sometimes are in the wrong order, depth or parent row.
2) Also there is an issue with scrollend, it triggers preemptively when the rows are expanded. I looks like scrollend is basing the end of the scroll base on the unexpanded rows
**To Reproduce**

a) Issue with appendData()
1. clone enterprise-wc, npm install
2. Replace tree-grid-virtual-scroll.ts with the attach file
[tree-grid-virtual-scroll.zip](https://github.com/infor-design/enterprise-wc/files/12291760/tree-grid-virtual-scroll.zip)
3. npm start
4. run http://localhost:4300/ids-data-grid/tree-grid-virtual-scroll.html
5. take note of the initial rows at the beginning
6. expand the first row
7. scroll down to the end
8. compare initial rows and after scroll
9. Error: the rows have changed order/depth or parent
10. open dev console
11. run document.querySelector("#tree-grid-virtual-scroll").collapseAll() and document.querySelector("#tree-grid-virtual-scroll").expandAll()
12. Error: TypeError: Cannot read properties of null (reading 'setAttribute')
image
Video:

https://github.com/infor-design/enterprise-wc/assets/53523076/2a94afdf-3402-4076-a454-28c1fcbbe4f0

b) Issue with scrollend trigger
2.1 Collapse all rows by setting all data to rowExpanded = false and  rowHidden: true
3.1 npm start
4.1 run http://localhost:4300/ids-data-grid/tree-grid-virtual-scroll.html
5.1 expand all expandable rows
6.1 scroll down 
7.1 error: scrollend triggers without going to the end of the scroll, this is more noticeable the more children is under a row

Video:

https://github.com/infor-design/enterprise-wc/assets/53523076/c8f69f2b-55ec-45ba-8b72-743082886afb

**Expected behavior**
a) The initial rows should not change, appendData() should only append the new rows.
b)  scrollend should trigger based on the expanded rows not the initially visible rows

**Version**

- ids-enterprise-wc beta 13

**Screenshots**
a) Initial rows and after scrolling to the end
image

**Platform**
Infor Application/Team Name: M3 UX
OS Version: Windows 11
Browser Name: chrome
Browser Version: 114.0.5735.248 (Official Build) (64-bit)

**Additional context**

### PR DESCRIPTION
**Pull request by clayinfor**
**Explain the details for making this change. What existing problem does the pull request solve?**

Adding new rows in the treegrid using appendData() was breaking rendering of existing rows. Current rows were sometimes are in the wrong order, depth or parent row.

Also there was an issue with `scrollend` where it triggered prematurely when the rows are expanded. It looks like `scrollend` was calculating based on unexpanded rows only.

Added an example that shows infinite-scrolling on a virtual-scroll enabled tree-grid.

**Related github/jira issue (required)**:

Closes #1425   

**Steps necessary to review your pull request (required)**:

1. Check out branch and run: `nvm use && npm run start`
2. Go to: http://localhost:4300/ids-data-grid/tree-grid-virtual-scroll-infinite-scroll.html
3. Take note of the initial rows at the beginning.
4. Expand the first row, then scroll down to the end.
5. Scroll back up to compare the initial rows after scrolling.
6. The rows should not  have changed order/depth or parent.
7. Open dev console and run `$("#tree-grid-virtual-scroll").collapseAll() && $("#tree-grid-virtual-scroll").expandAll()`
8. No errors should be thrown.
9. Also the `scrollend` event should continue to fire at the right time. 

**Included in this Pull Request**:
- [ ] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

**Comment by jdilag90** [comment]
there's an issue where the collapsed state of the rows change after an infinite scroll

1) Go to http://localhost:4300/ids-data-grid/tree-grid-virtual-scroll-infinite-scroll.html
2) the first row is in a collapsed state on init
3) scroll down to trigger an infinite scroll
4) scroll up to see the first row is now expanded (the icon also still reflects a collapsed state)

**Comment by clayinfor** [comment]
Thanks @jdilag90 , the expand-icons are fixed now.

**Comment by jdilag90** [comment]
looks good, one more thing
you can scroll to row 300 in our current infinite-scroll example, you can't in this PR

https://main.wc.design.infor.com/ids-data-grid/infinite-scroll.html
vs.
http://localhost:4300/ids-data-grid/infinite-scroll.html

File: doc/CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
 - `[DataGrid]` Fixes scroll jumping in virtual/infinite scroll. ([#1390](https://github.com/infor-design/enterprise-wc/issues/1390))
 - `[DataGrid]` Fix scrollend event triggering in different zoom levels. ([#1396](https://github.com/infor-design/enterprise-wc/issues/1396))
 - `[DataGrid]` Added a `beforerowselected` and `beforeredeselected` event that can be vetoed to both lookup and datagrid. ([#1304](https://github.com/infor-design/enterprise-wc/issues/1304))
+- `[DataGrid]` Treegrid `appendData()` no longer breaks rendering of existing rows, and `scrollend` triggers properly. ([#1425](https://github.com/infor-design/enterprise-wc/issues/1425))
```
Comment (tmcconechy) [inline doc/CHANGELOG.md:22]: Seems to be in the wrong version?

**Review by tmcconechy — COMMENTED** [review]
Everything looks good to me aside from what @jdilag90 reported. Just a minor thing in the change log

File: doc/CHANGELOG.md (hunk 1 above)
Comment (clayinfor) [reply doc/CHANGELOG.md]: Hey @tmcconechy , I have updated the changelog and fixed the issue @jdilag90 mentioned above.

**Review by tmcconechy — APPROVED** [review]

**Review by tmcconechy — APPROVED** [review]

**Review by jdilag90 — APPROVED** [review]

**Pull request merged**